app/src/main/java/org/mozilla/focus/utils/UrlUtils.kt

```diff
@@ -0,0 +1,192 @@
+                scheme.startsWith("file") ||
+                scheme.startsWith("data") ||
+                scheme.startsWith("javascript") ||
+                scheme.startsWith("about"))
+    }
+
+    fun isSupportedProtocol(scheme: String?): Boolean {
+        return scheme != null && (isPermittedResourceProtocol(scheme) || scheme.startsWith("error"))
+    }
+
+    fun isInternalErrorURL(url: String): Boolean {
+        return "data:text/html;charset=utf-8;base64," == url
+    }
+
+    /**
+     * Checks that urls are non-null and are the same aside from a trailing slash.
+     *
+     * @return true if urls are the same except for trailing slash, or if either url is null.
+     */
+    fun urlsMatchExceptForTrailingSlash(url1: String?, url2: String?): Boolean {
+        // This is a hack to catch a NPE in issue #26.
+        if (url1 == null || url2 == null) {
+            return false
+        }
+
+        val lengthDifference = url1.length - url2.length
+
+        return when (lengthDifference) {
+            0 -> // The simplest case:
+                url1.equals(url2, ignoreCase = true)
+            1 -> // url1 is longer:
+                url1[url1.length - 1] == '/' && url1.regionMatches(0, url2, 0, url2.length, ignoreCase = true)
+            -1 -> url2[url2.length - 1] == '/' && url2.regionMatches(0, url1, 0, url1.length, ignoreCase = true)
+            else -> false
+        }
+    }
+
+    @Suppress("MagicNumber")
+    fun stripCommonSubdomains(host: String?): String? {
+        if (host == null) {
+            return null
+        }
+
+        // In contrast to desktop, we also strip mobile subdomains,
+        // since its unlikely users are intentionally typing them
+        var start = 0
+
+        when {
+            host.startsWith("www.") -> start = 4
+            host.startsWith("mobile.") -> start = 7
+            host.startsWith("m.") -> start = 2
+        }
+
+        return host.substring(start)
+    }
+
+    @Suppress("MagicNumber")
+    fun stripScheme(url: String?): String? {
+        if (url == null) {
+            return null
+        }
+
+        var start = 0
+        when {
+            url.startsWith("http://") -> start = 7
+            url.startsWith("https://") -> start = 8
+        }
+
+        return url.substring(start)
+    }
+
+    fun stripSchemeAndSubDomain(url: String): String {
+        return normalize(stripCommonSubdomains(stripScheme(url))!!)
+    }
+
+    fun isLocalizedContent(url: String?): Boolean {
+        return url != null &&
+                (url == LocalizedContent.URL_ABOUT ||
+                        url == LocalizedContent.URL_RIGHTS ||
+                        url == "about:blank"
+                        )
+    }
+
+    fun getIndexOfTLD(url: String): Int {
+        val tld = getTopLevelDomain(url) ?: ""
+        return url.indexOf(tld, 0, true) + tld.length
+    }
+
+    fun getTopLevelDomain(url: String): String? {
+        return TLDExtractor.extractTLD(URL(url).host)
+    }
```

Comment:
nit: Those methods seem to deal with the "public suffix" which is not the top level domain in all cases. For example `blogspot.com` is a public suffix (because users can register a domain under that suffix) while the TLD is just "com".